### PR TITLE
Change error handling, and check for blank proxy names

### PR DESCRIPTION
**Pull request by xthexder**
Fixes https://github.com/Shopify/toxiproxy/issues/51 and gives better errors back form the go api

@Sirupsen

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -103,6 +104,15 @@ func (server *server) ProxyCreate(response http.ResponseWriter, request *http.Re
 		return
 	}
 
+	if len(input.Name) < 1 {
+		http.Error(response, server.apiError(errors.New("Missing required field: name"), http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+	if len(input.Upstream) < 1 {
+		http.Error(response, server.apiError(errors.New("Missing required field: name"), http.StatusBadRequest), http.StatusBadRequest)
```
Comment (Sirupsen) [inline api.go]: name => upstream, we need listen as well perhaps?

File: api_test.go [hunk 2]
```diff
@@ -61,6 +61,18 @@ func TestCreateProxy(t *testing.T) {
 	})
 }
 
+func TestCreateProxyBlankName(t *testing.T) {
```
Comment (Sirupsen) [inline api_test.go]: Test for upstream

**Comment by Sirupsen** [comment]
oh wow thanks! I just left that issue there as a note to myself to fix it :heart:

File: api.go (hunk 1 above)
Comment (xthexder) [inline api.go]: A blank listen field means use an ephemeral port, so that's allowed.

Comment (Sirupsen) [inline api.go]: Oh yes!

**Comment by Sirupsen** [comment]
LGTM

**Comment by Sirupsen** [comment]
bump :)

**Pull request merged**